sre_yield/__init__.py

```diff
@@ -563,14 +572,22 @@ def __init__(self, pattern, flags=0, charset=CHARSET, max_count=None):
             sre_constants.MIN_REPEAT: self.max_repeat_values,
             sre_constants.MAX_REPEAT: self.max_repeat_values,
             sre_constants.AT: self.nothing_added,
-            sre_constants.ASSERT: self.empty_list,
-            sre_constants.ASSERT_NOT: self.empty_list,
+            sre_constants.ASSERT: self.lookaround_parse_error,
+            sre_constants.ASSERT_NOT: self.lookaround_parse_error,
             sre_constants.ANY: lambda _: self.in_values(((sre_constants.NEGATE,),)),
             sre_constants.IN: self.in_values,
             sre_constants.NOT_LITERAL: self.not_literal,
             sre_constants.CATEGORY: self.category,
             sre_constants.GROUPREF: self.groupref,
         }
+        if self.relaxed:
+            self.backends.update(
+                {
+                    sre_constants.ASSERT: self.nothing_added,
```

Comment:
I needed to use `self.maybe_save` here, but only for my test cases, not for my real usage.  This patch solves my real usage perfectly, so big :+1: from me.

I'll dig into this a bit later today to understand why I needed `self.maybe_save` and why your patch doesn't, but feel free to merge if you already know the answer ;-).